walk/walk.go

```diff
@@ -155,31 +154,42 @@ func visit(c *config.Config, cexts []config.Configurer, isBazelIgnored isIgnored
 	c = configure(cexts, knownDirectives, c, rel, f)
 	wc := getWalkConfig(c)
 
-	if wc.isExcluded(rel, ".") {
+	// Don't recurse into excluded directories
+	if wc.isExcluded(rel) {
 		return
 	}
 
 	var subdirs, regularFiles []string
 	for _, ent := range ents {
-		base := ent.Name()
-		if isBazelIgnored(path.Join(rel, base)) || wc.isExcluded(rel, base) {
+		name := ent.Name()
+		entRel := path.Join(rel, name)
+
+		// Skip blank entry names, .git directories, ignored and excluded.
+		if name == "" || name == ".git" || isBazelIgnored(entRel) || wc.isExcluded(entRel) {
 			continue
 		}
-		ent := resolveFileInfo(wc, dir, rel, ent)
+
+		// Skip git-ignored when the gitignore option is enabled.
+		if wc.gitignoreEnabled && wc.isGitIgnored(append(relParts, name), ent.IsDir()) {
```

Comment:
I think you want to do `relParts = append(relParts, name)` and then `relParts = relParts[:len(relParts)-1]` after the call. Otherwise when `relParts` is full you are going to realloc it for every ent

Reply:
Good idea for this loop.

Note for the recursive loop below I don't think we should though. Technically we probably can atm but that might get confusing.